ml-proto/test/block.wast

```diff
@@ -200,7 +200,7 @@
 
 ;; TODO(stack): move this elsewhere
 (module (func $type-break-num-vs-void
-  (block (i32.const 66) (br 0))
+  (block (drop (i32.const 66)) (br 0))
```

Comment:
As far as I am aware the 'explicit drop' constraints do not make these examples invalid, it's just binaryen, and we seem to agree that some of these are not incompatible with an AST either. I don't think these spec tests should be changed just to follow a more restrictive interpretation being made in binaryen.

The tests that would be incompatible with the binaryen AST for now are those that use an ordering that would demand `first` or lexical scoped constants. I think binaryen should work with the rest, relax the constraints, and just separate the remainder.